streams/src/main/java/org/apache/kafka/streams/kstream/internals/WindowedDeserializer.java

```diff
@@ -17,25 +17,53 @@
 
 package org.apache.kafka.streams.kstream.internals;
 
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.config.ConfigException;
 import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.utils.Utils;
 import org.apache.kafka.streams.kstream.Windowed;
 
 import java.nio.ByteBuffer;
 import java.util.Map;
 
+/**
+ *  The inner deserializer class can be specified by setting the property key.deserializer.inner.class,
+ *  value.deserializer.inner.class or deserializer.inner.class,
+ *  if the no-arg constructor is called and hence it is not passed during initialization.
+ *  Note that the first two take precedence over the last.
+ */
 public class WindowedDeserializer<T> implements Deserializer<Windowed<T>> {
 
     private static final int TIMESTAMP_SIZE = 8;
 
     private Deserializer<T> inner;
 
+    // Default constructor needed by Kafka
+    public WindowedDeserializer() {}
+
     public WindowedDeserializer(Deserializer<T> inner) {
         this.inner = inner;
     }
 
     @Override
     public void configure(Map<String, ?> configs, boolean isKey) {
-        // do nothing
+        if (inner == null) {
+            String propertyName = isKey ? "key.deserializer.inner.class" : "value.deserializer.inner.class";
+            Object innerDeserializerClass = configs.get(propertyName);
+            propertyName = (innerDeserializerClass == null) ? "deserializer.inner.class" : propertyName;
+            String value = null;
+            try {
+                value = (String) configs.get(propertyName);
+                Object o = Utils.newInstance(value, Deserializer.class);
+                if (!Deserializer.class.isInstance(o)) {
```

Comment:
Agreed. However, the check is introduced by AbstractConfig.getConfiguredInstance. Do you think we should just keep the same as it or remove this unnecessary check only for two classes?